### PR TITLE
fix(formatter/member-chain): regex formatting

### LINKED ISSUE #300: 🐛 Incorrect formatting in function chain
### Environment information

```block
main
```

### What happened?

Formatting for function chaining is [wonky and different from prettier](https://biomejs.dev/playground/?indentStyle=space&quoteStyle=single&jsxQuoteStyle=single&lintRules=all&code=YwBvAG4AcwB0ACAAYQAgAD0AIAB7AAoAIAAgAGwAbwBjAGEAbABlAHMAOgAgAHoALgByAGUAYwBvAHIAZAAoAAoAIAAgACAAIABsAG8AYwBhAGwAZQBLAGUAeQBTAGMAaABlAG0AYQAsAAoAIAAgACAAIAB6AC4AcwB0AHIAaQBuAGcAKAApAC4AbQBpAG4AKAAyACkALgByAGUAZwBlAHgAKAAvAF4AWwBhAC0AegBBAC0AWgBcAC0AXQArACQALwBnAG0ALAAgAHsACgAgACAAIAAgACAAIABtAGUAcwBzAGEAZwBlADoAIAAnAE8AbgBsAHkAIABFAG4AZwBsAGkAcwBoACAAYQBsAHAAaABhAGIAZQB0ACAAcwB5AG0AYgBvAGwAcwAgAGEAbgBkACAAaAB5AHAAaABlAG4AIABhAGwAbABvAHcAZQBkACcALAAKACAAIAAgACAAfQApAAoAIAAgACkALAAKAH0A) 

### Expected result

It should correctly format function chaining

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by victor-teles**
## Summary

This PR fixes #300

### About the resolution

After some research in prettier repo, I discovered that regex expressions should only be considered `simple` if they have a length less than or equal to 5

Prettier ref: https://github.com/prettier/prettier/blob/d4d4b185e0ddc3e0dd839b873e2ee7fe8131b684/src/language-js/utils/index.js#L761C1-L761C1

## Test Plan

- Validate snapshot 😄
- Added some regex cases

### CODE REVIEW

**Review by denbezrukov — APPROVED** [review]

File: crates/biome_js_formatter/src/utils/member_chain/simple_argument.rs [hunk 1]
```diff
@@ -161,6 +163,19 @@ impl SimpleArgument {
         }
     }
 
+    fn is_simple_regex_expression(&self) -> bool {
```
Comment (ematipico) [inline crates/biome_js_formatter/src/utils/member_chain/simple_argument.rs:166]: Let's add a documentation comment that explains **when** a regex is simple.

While you added this information to the PR, that comment will be lost; instead, having it as a comment is helpful to future contributors.

**Review by ematipico — APPROVED** [review]
Looks good to me, just left a minor comment. Feel free to merge the PR once addressed.

**Review by Conaclos — APPROVED** [review]
> After some research in prettier repo, I discovered that regex expressions should only be considered simple if they have a length less than or equal to 5

:o this seems so strange as behavior!

Good work, everything looks great :)

**Comment by ematipico** [comment]
> > After some research in prettier repo, I discovered that regex expressions should only be considered simple if they have a length less than or equal to 5
> 
> :o this seems so strange as behavior!
> 

That's prettier for you :)

**Pull request merged**